notebooks/article_1.py

```diff
@@ -0,0 +1,1093 @@
+num_variables = len(specific_variables)
+num_columns = 5
+num_rows = int(np.ceil(num_variables / num_columns))
+
+# Set the overall figure size
+fig_width = num_columns * 6
+fig_height = num_rows * 5
+
+plt.figure(figsize=(fig_width, fig_height))
+
+for i, column in enumerate(specific_variables):
+    plt.subplot(num_rows, num_columns, i+1)
+    plt.title(column, fontsize=25)
+    sns.histplot(dataframe[column], color="red", kde=True)  # Adding KDE for smooth distribution curve.
+    plt.xlabel('')
+    plt.ylabel('')
+
+# Adjust layout to prevent overlap
+plt.tight_layout()
+plt.show()
+
+
+# As observed in the histogram, noticeable spikes are apparent in the data. Excluding binary variables, the variables with the most prominent spikes are:
+#
+# - `age_first_funding_year`
+# - `age_first_milestone_year`
+# - `age_last_milestone_year`
+# - `funding_total_usd`
+# - `relationships`
+
+# Now that we have identified outliers in our data, the next step is normalization. We will utilize the log-transformed method for this purpose.
+
+# #### Handle Outliers
+
+# To handle this outliers we will use **The log-transformed method** that involves taking the natural logarithm (ln) of each data point in a dataset. It's useful when data has a wide range of values, helping to balance the data.
+#
+# The following graph illustrates the dataset before and after applying log transformation.
+
+# In[608]:
+
+
+age_features = ["age_first_funding_year", "age_last_funding_year", "age_first_milestone_year", "age_last_milestone_year"]
+
+# Create a figure with a specific size
+plt.figure(figsize=(17, 6), dpi=100)
+
+# Loop through the list of variables
+for i, variable in enumerate(age_features):
+    # Regular histogram
+    plt.subplot(2, 4, i + 1)
+    sns.histplot(dataframe[variable], color="red", kde=True)
+    plt.title(variable)
+
+    # Log-transformed histogram
+    plt.subplot(2, 4, i + 5)
+    log_variable = np.log(dataframe[variable] + 1)  # Adding 1 to avoid log(0)
+    sns.histplot(log_variable, color="blue", kde=True)
+    plt.title(f"log_{variable}")
+
+plt.tight_layout()
+plt.show()
+
+
+# The distinction is evident; in the blue histogram, the peak is noticeably lower than in the red histogram.
+#
+# When you use the logarithm on data, **compresses** the outliers closer to the rest of the values. This makes the distribution of the data look more like a balanced and symmetrical shape, similar to a bell curve.
+#
+# It's important to note that log transformation doesn't remove outliers entirely; it reduces their impact.
+#
+# Now that we've addressed negative values and outliers, lets quickly inspect the first row of your DataFrame after sorting by index.
+
+# In[609]:
+
+
+dataframe.sort_index().head(3)
+
+
+# ### Creating Variables
+
+# In this part, we'll demonstrate how useful it is to **create temporary variables**. The benefits of using temporary variables include ensuring the integrity of the original data, particularly during complex operations.
```

Comment:
Can you please rework this entire paragraph? Instead of "we'll demonstrate how useful it is to create temporary variables".
Say instead that it can be useful to create new feature from the original date such as "age" or founding year.
Please explain why.